### PR TITLE
Webpack builds!

### PR DESCRIPTION
**Pull request by gesa**
### Summary

TODO:
- [x] add eslint runner
- [x] set up dev server
- [x] setup a jest file transformer for html imports (webpack doesn't support the fs module, but does have an html loader)
- [x] create dev, prod, & shared webpack config

### Checklist

- [ ] Added a changelog entry

- @gesa

### CODE REVIEW

File: test/app/index.html [hunk 1]
```diff
@@ -229,7 +229,7 @@ <h1>Braintree Drop-in demo</h1>
     results.textContent = '';
     error.textContent = '';
 
-    braintree.dropin.create(config, function (err, instance) {
+    dropin.create(config, function (err, instance) {
```
[the remark this answers is not in the document]
Comment (crookedneighbor) [reply test/app/index.html:232]: I think it's pretty important to keep everything name-spaced under `braintree`. Surely there's a way to do that, right?

File: src/dropin.js [hunk 2]
```diff
@@ -734,7 +733,7 @@ Dropin.prototype._injectStylesheet = function () {
 
   if (document.getElementById(constants.STYLESHEET_ID)) { return; }
 
-  stylesheetUrl = ASSETS_URL + '/web/dropin/' + VERSION + '/css/dropin@DOT_MIN.css';
+  stylesheetUrl = ASSETS_URL + '/web/dropin/' + VERSION + '/css/dropin.css';
```
[the remark this answers is not in the document]
Comment (crookedneighbor) [reply src/dropin.js]: Can you clarify this? The intention before was to load the minified css from the cdn when using the minified version on the cdn, and the un-minified version when using the un-minified version.

I do wonder now if folks using the build from npm always got the un-minified css assets.

File: package.json [hunk 3]
```diff
@@ -53,19 +58,33 @@
...
     "mkdirp": "^1.0.3",
     "mocha": "^7.1.0",
     "npm-run-all": "^4.1.5",
+    "postcss-clean": "^1.1.0",
+    "postcss-loader": "^3.0.0",
+    "postcss-preset-env": "^6.7.0",
     "serve-static": "^1.14.1",
+    "string-replace-loader": "^2.2.0",
+    "style-loader": "^1.1.3",
+    "symlink-webpack-plugin": "github:gesa/symlink-webpack-plugin#e661f2d64590a3cacba42d4f3c8a36f8d6dff1e4",
```
[the remark this answers is not in the document]
Comment (crookedneighbor) [reply package.json:79]: Did this ever get released?

File: package.json [hunk 4]
```diff
@@ -37,13 +37,18 @@
...
+    "clean-webpack-plugin": "^3.0.0",
     "connect": "^3.7.0",
+    "copy-webpack-plugin": "^5.1.1",
+    "css-loader": "^3.4.2",
     "del": "^5.1.0",
     "dotenv": "^8.2.0",
     "es6-shim": "^0.35.5",
     "eslint": "^6.8.0",
     "eslint-config-braintree": "^4.0.0",
+    "eslint-loader": "^3.0.3",
```
[the remark this answers is not in the document]
Comment (crookedneighbor) [reply package.json:49]: Could always run it in webpack and have CI run a `npm run lint` task directly that is _not_ a pre or post task for tests.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (gesa) [reply package.json:79]: looks like it https://github.com/hmsk/symlink-webpack-plugin/releases/tag/v1.0.0
```suggestion
    "symlink-webpack-plugin": "^1.0.0",
```

**Comment by armandodlvr** [comment]
closing this PR in favor of https://github.com/braintree/braintree-web-drop-in/pull/844

**Comment by shango420** [comment]
> closing this PR in favor of https://github.com/braintree/braintree-web-drop-in/pull/844

**Pull request closed without merging**